jetson/nav/rover.cpp

```diff
@@ -242,6 +261,47 @@ bool Rover::updateRover( RoverStatus newRoverStatus )
             mRoverStatus.odometry() = newRoverStatus.odometry();
             mRoverStatus.leftTarget() = newRoverStatus.leftTarget();
             mRoverStatus.rightTarget() = newRoverStatus.rightTarget();
+
+            // Perform basic caching, include special case when
+            // we store for the first time, but store an empty one
+            // so we don't just have "junk" in the target Cache.
+            if ( mRoverStatus.leftTarget().distance != -1 ) 
+            {
+                mRoverStatus.leftCacheTarget() = mRoverStatus.leftTarget();
+                mRoverStatus.getMisses() = 0;
+            }
+            else 
+            { // if this is empty, then so is rightTarget. increment misses
+                mRoverStatus.getMisses()++;
+            }
+
+            // Note: The only time this updates is ONLY if the leftTarget updates (by default
+            // the first target seen will be stored in leftTarget, and even if only 1 is seen
+            // this won't update, so we should be fine)
+            if ( mRoverStatus.rightTarget().distance != -1 ) 
+            {
+                mRoverStatus.rightCacheTarget() = mRoverStatus.rightTarget();
+                mRoverStatus.getMisses() = 0;
+            }
+
+
+            // If we have too many misses in a row, then we know we definitely are missing the Tag,
+            // so we can set the cache to the left/right target (which both should be empty)
+            // Must also do the reset once we finish a gate search
+            if ( mRoverStatus.getMisses() > mRoverConfig[ "navThresholds" ][ "cacheMissMax" ].GetDouble() )
+            {
+                mRoverStatus.getMisses() = 0;
+                // these will be zero, so set them as such since we know
+                // they definitely aren't there
+                mRoverStatus.leftCacheTarget() = mRoverStatus.leftTarget();
+                mRoverStatus.rightCacheTarget() = mRoverStatus.rightTarget();
+            }
+
+
+            cout << "LEFT CACHE (distance, bearing): (" << mRoverStatus.leftCacheTarget().distance << "," << mRoverStatus.leftCacheTarget().bearing << ")" << endl;
```

Comment:
Are these cout statements just for testing?

Reply:
My bad, I'll make sure to remove them (wanted to keep them in for when we would test in-person, but I understand that it's better to remove them for style)